modules/smaatoBidAdapter.js

```diff
@@ -180,6 +180,10 @@ export const spec = {
           }
         }
         resultingBid.meta.mediaType = resultingBid.mediaType;
+
+        if (bid.ext?.dsa) {
```

Comment:
wdyt about placing this above if/else/switch in line 148?
because i find it always weird that we push resultingBid and then later change it again (for me it makes more sense to build it, then push the complete thing)

or alternatively adding it conditionally directly at the beginning together with the other values.
i think it should also work like this and not break the tests with complaints about dsa being undefined,  but please check

```
let resultingBid = {
          requestId: bid.impid,
          cpm: bid.price || 0,
          width: bid.w,
          height: bid.h,
          ttl: ttlInSec,
          creativeId: bid.crid,
          dealId: bid.dealid || null,
          netRevenue: deepAccess(bid, 'ext.net', true),
          currency: CURRENCY,
          meta: {
            advertiserDomains: bid.adomain,
            networkName: bid.bidderName,
            agencyId: seatbid.seat,
            ...(bid.ext?.dsa && {dsa: bid.ext.dsa})
          }
        };
```



but if you dislike both it's also fine for me like this

Reply:
> or alternatively adding it conditionally directly at the beginning together with the other values.
I like inline things, when it makes sense of courses, which is the case here 😄  Tests are passing, so it works.

> because i find it always weird that we push resultingBid and then later change it again (for me it makes more sense to build it, then push the complete thing)
Completely agree. I would move pushing the resultingBid after the line 183, means push once at the end when the whole object built. And then there is the `default` case in the switch where we could use `return` which should be equivalent of `continue` in java. 
I saw in your team old PRs to prebid that prebid people do not like refactoring in the scope of features, so I did nothing.